compiler/kernel/op.go

```diff
@@ -588,28 +588,42 @@ func (b *Builder) compileTrunk(trunk *dag.Trunk, parent zbuf.Puller) ([]zbuf.Pul
 			if err != nil {
 				return nil, err
 			}
+			var pruner expr.Evaluator
+			if trunk.KeyPruner != nil {
+				pruner, err = compileExpr(trunk.KeyPruner)
+				if err != nil {
+					return nil, err
+				}
+			}
```

Comment:
Nit: This also appears immediately after the enclosing if statement, so you can delete both copies in favor of one placed before the if statement.

Reply:
That doesn't work because the lister and the scanner need their own copies as Evaluators can't be used concurrently.

Reply:
👍